frontend/src/metabase/query_builder/selectors.unit.spec.js

```diff
@@ -133,6 +142,41 @@ describe("getIsResultDirty", () => {
       expect(getIsResultDirty(state)).toBe(true);
     });
 
+    it("converts clauses into plain MBQL objects", () => {
+      const aggregation = ["count"];
+      const breakout = ["field", ORDERS.CREATED_AT.id, null];
+      const filter = [">=", ["field", ORDERS.TOTAL.id, null], 20];
+      const orderBy = ["asc", ["aggregation", 0]];
+      const join = {
+        alias: "Products",
+        condition: [
+          "=",
+          ["field", ORDERS.PRODUCT_ID.id, null][
+            ("field", PRODUCTS.ID.id, null)
+          ],
+        ],
+      };
```

Comment:
Could you please clarify what does the condition mean here? 

```js
["field", ORDERS.PRODUCT_ID.id, null][
  ("field", PRODUCTS.ID.id, null)
]
```

I'm not sure I've seen anything like that in MBQL before